File: 2nd_quarter/week09/Rash/부족한금액 계산하기.py
#풀이계획
#1. totalPrice 변수를 선언
#2. tmpPrice로 계속 올라가는 변수 저장해놓기
#3. totalPrice값을 구해서 Money를 빼면 부족한 금액이 나온다.
#4. 만약 totalPrice 보다 money가 작으면 0을 return해라

def solution(price, money, count):
    totalPrice = 0
    tmpPrice = 0
    for i in range(count):
        tmpPrice += price
        totalPrice += tmpPrice
    if totalPrice < money:
        return 0
    answer = totalPrice - money
    return answer

File: 2nd_quarter/week09/Whaeun/부족한 금액 계산하기.cpp
#include <iostream>
#include <map>
#include <algorithm>


using namespace std;

long long solution(int price, int money, int count)
{
    long long answer = 0;
    answer = money;
    
    for(int i = 1; i <= count; i++){
           answer -= (price *i);
    }
    
    if(answer > 0){
        answer = 0;
    }
    else{
        answer *= -1;
    }

    return answer;
}
